### PR TITLE
Enable assimp-fuzzing test

### PR DESCRIPTION
**Pull request by kimkulling**
- enable assimp fuzzer

### CODE REVIEW

**Comment by inferno-chromium** [comment]
Please check the presubmit and other job failures above.

```
[367/367] Linking CXX executable bin/unit
+ clang++ -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++ -std=c++11 -I. assimp/fuzz/assimp_fuzzer.cc -o /out/assimp_fuzzer -fsanitize=fuzzer -lassimp
clang-11: error: no such file or directory: 'assimp/fuzz/assimp_fuzzer.cc'
Building fuzzers failed.
Running: docker build -t gcr.io/oss-fuzz/assimp projects/assimp
WARNING: language not specified in project.yaml. Build may fail.
Keeping existing build artifacts as-is (if any).
Running: docker run --rm --privileged --cap-add SYS_PTRACE -e FUZZING_ENGINE=libfuzzer -e SANITIZER=address -e ARCHITECTURE=x86_64 -v /home/runner/work/oss-fuzz/oss-fuzz/build/out/assimp:/out -v /home/runner/work/oss-fuzz/oss-fuzz/build/work/assimp:/work -t gcr.io/oss-fuzz/assimp
Building project assimp
Running command: python /home/runner/work/oss-fuzz/oss-fuzz/infra/helper.py build_fuzzers assimp --engine libfuzzer --sanitizer address --architecture x86_64
Failed projects: assimp
##[error]Process completed with exit code 1.
```

**Comment by kimkulling** [comment]
Shall be fixed now.

**Comment by kimkulling** [comment]
Hi,
the build is fixed. At the moment the dataflow sanitizer check of your CI failes with a bundle of unresolved symbols:

```
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp3IFC10Schema_2x328IfcElectricDistributionPointD2Ev':
/src/assimp/code/AssetLib/IFC/IFCReaderGen_2x3.h:2361: undefined reference to `dfs$_ZNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEED2Ev'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp4STEP12ObjectHelperINS_3IFC10Schema_2x326IfcCableCarrierSegmentTypeELm1EED0Ev':
/src/assimp/code/AssetLib/Step/STEPFile.h:471: undefined reference to `dfs$_ZdlPv'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp3IFC10Schema_2x326IfcCableCarrierSegmentTypeD0Ev':
/src/assimp/code/AssetLib/IFC/IFCReaderGen_2x3.h:1840: undefined reference to `dfs$_ZdlPv'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp3IFC10Schema_2x326IfcCableCarrierSegmentTypeD2Ev':
/src/assimp/code/AssetLib/IFC/IFCReaderGen_2x3.h:1840: undefined reference to `dfs$_ZNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEED2Ev'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp4STEP12ObjectHelperINS_3IFC10Schema_2x319IfcWallStandardCaseELm0EED0Ev':
/src/assimp/code/AssetLib/Step/STEPFile.h:471: undefined reference to `dfs$_ZdlPv'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp3IFC10Schema_2x319IfcWallStandardCaseD0Ev':
/src/assimp/code/AssetLib/IFC/IFCReaderGen_2x3.h:3854: undefined reference to `dfs$_ZdlPv'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp4STEP12ObjectHelperINS_3IFC10Schema_2x311IfcCsgSolidELm1EED0Ev':
/src/assimp/code/AssetLib/Step/STEPFile.h:471: undefined reference to `dfs$_ZdlPv'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp3IFC10Schema_2x311IfcCsgSolidD0Ev':
/src/assimp/code/AssetLib/IFC/IFCReaderGen_2x3.h:2160: undefined reference to `dfs$_ZdlPv'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp4STEP12ObjectHelperINS_3IFC10Schema_2x311IfcBeamTypeELm1EED0Ev':
/src/assimp/code/AssetLib/Step/STEPFile.h:471: undefined reference to `dfs$_ZdlPv'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o):/src/assimp/code/AssetLib/IFC/IFCReaderGen_2x3.h:1709: more undefined references to `dfs$_ZdlPv' follow
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp3IFC10Schema_2x311IfcBeamTypeD2Ev':
/src/assimp/code/AssetLib/IFC/IFCReaderGen_2x3.h:1709: undefined reference to `dfs$_ZNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEED2Ev'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp4STEP12ObjectHelperINS_3IFC10Schema_2x321IfcAnnotationFillAreaELm2EED0Ev':
/src/assimp/code/AssetLib/Step/STEPFile.h:471: undefined reference to `dfs$_ZdlPv'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp3IFC10Schema_2x321IfcAnnotationFillAreaD0Ev':
/src/assimp/code/AssetLib/IFC/IFCReaderGen_2x3.h:1565: undefined reference to `dfs$_ZdlPv'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp4STEP12ObjectHelperINS_3IFC10Schema_2x331IfcStructuralCurveMemberVaryingELm0EED0Ev':
/src/assimp/code/AssetLib/Step/STEPFile.h:471: undefined reference to `dfs$_ZdlPv'
/src/assimp/lib/libassimp.a(IFCReaderGen1_2x3.cpp.o): In function `dfs$_ZN6Assimp3IFC10Schema_2x331IfcStructuralCurveMemberVaryingD0Ev':

```
I can reproduce the issue on my local machine but I do not have a clue how to fix this issue. Any idea?

**Comment by kimkulling** [comment]
Ok, I fixed it. Ready for merging.

File: projects/assimp/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER kim.kulling@googlemail.com
```
Comment (inferno-chromium) [inline projects/assimp/Dockerfile:18]: Remove MAINTAINER field, not needed anymore.

File: projects/assimp/build.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# show config
```
Comment (inferno-chromium) [inline projects/assimp/build.sh:18]: Remove these echos, we do echo the flags in compile

File: projects/assimp/build.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# generate build env and build assimp
+cmake CMakeLists.txt -G "Ninja" -DBUILD_SHARED_LIBS=OFF -DASSIMP_BUILD_ZLIB=ON -DASSIMP_BUILD_TESTS=OFF -DASSIMP_BUILD_ASSIMP_TOOLS=OFF -DASSIMP_BUILD_SAMPLES=OFF
+cmake --build .
+
+# build the fuzzer
+$CXX $CXXFLAGS -std=c++11 -I$SRC/assimp/include \
+	fuzz/assimp_fuzzer.cc -o $OUT/assimp_fuzzer \
+	$LIB_FUZZING_ENGINE $SRC/assimp/lib/libassimp.a $SRC/assimp/lib/libIrrXML.a $SRC/assimp/lib/libzlibstatic.a
+
+
```
Comment (inferno-chromium) [inline projects/assimp/build.sh:35]: remove extra new line.

File: projects/assimp/project.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+homepage: "https://github.com/assimp/assimp"
+language: c++
+primary_contact: "kim.kulling@googlemail.com"
+auto_ccs:
+  - "kim.kulling@googlemail.com"
+#fuzzing_engines:
```
Comment (inferno-chromium) [inline projects/assimp/project.yaml:6]: remove these commented lines.

File: projects/assimp/Dockerfile (hunk 1 above)
Comment (kimkulling) [reply projects/assimp/Dockerfile]: Fixed

File: projects/assimp/build.sh (hunk 3 above)
Comment (kimkulling) [reply projects/assimp/build.sh]: Fixed

**Pull request merged**